server/src/main/java/org/opensearch/cluster/decommission/DecommissionAttributeMetadata.java

```diff
@@ -79,35 +79,29 @@ public DecommissionStatus status() {
     /**
      * Returns instance of the metadata with updated status
      * @param newStatus status to be updated with
-     * @return instance with valid status
      */
     // synchronized is strictly speaking not needed (this is called by a single thread), but just to be safe
-    public synchronized DecommissionAttributeMetadata setUpdatedStatus(DecommissionStatus newStatus) {
-        // if the current status is the expected status already, we return the same instance
-        if (newStatus.equals(status)) {
-            return this;
+    public synchronized void validateNewStatus(DecommissionStatus newStatus) {
+        // if the current status is the expected status already or new status is FAILED, we let the check pass
+        if (newStatus.equals(status) || newStatus.equals(DecommissionStatus.FAILED)) {
+            return;
```

Comment:
This is assuming that all steps can have a self-loop for state transitions

Reply:
Didn't get this. This is added to ensure that if any step wants to mark the status as `FAILED` during decommission, we will allow it to do so. Let me know if you have any specific case in mind

Reply:
And this behaviour is same as before. Had to refactor this method a bit because we were updating the same instance which led to relative diff as 0 for decommission state update

Reply:
The self loops which we might come up during multiple concurrent requests is handled seperately as part of this PR https://github.com/opensearch-project/OpenSearch/pull/4684. Today we need status to get into same state as per current service implementation

Reply:
The other condition makes it look paranoid